src/hera/workflows/script.py

```diff
@@ -156,31 +157,66 @@ def _build_inputs(self) -> Optional[ModelInputs]:
             else:
                 func_parameters = _get_parameters_from_callable(self.source)
 
-        if inputs is None and not func_parameters and not func_artifacts:
+        return self._process_io(inputs, func_parameters, func_artifacts, False)
+
+    def _build_outputs(self) -> Optional[ModelOutputs]:
+        outputs = super()._build_outputs()
+
+        if not callable(self.source):
+            return outputs
+
+        if not global_config.experimental_features["script_annotations"]:
+            return outputs
+
+        out_parameters, out_artifacts = _get_outputs_from_callable(self.source)
+        func_parameters, func_artifacts = _get_outputs_from_callable_signature(self.source)
+        func_parameters.extend(out_parameters)
+        func_artifacts.extend(out_artifacts)
+
+        return self._process_io(outputs, func_parameters, func_artifacts, True)
+
+    def _process_io(
```

Comment:
What do you think abut moving this logic to `hera.workflows._mixins.IOMixin`? I think the IO can be constructed there and then the `Script` remains free of that logic / maintains only the `build`ers

Reply:
I believe this logic is unique for scripts as it deals with extracting information from the function signature. If we ever want to reuse that we could possibly create a `FunctionIOMixin` containing that logic. What do you think?

Reply:
Sounds good to me 👍🏻